### ISSUE TITLE
How to ensure that the Expected.txt file also gets updated when the duke.txt file is changed?

### ISSUE THREAD

**Issue by MadLamprey** [nus-cs2103-AY2324S1/forum#32]
Since the outputs [like size of array list and when list gets displayed] depend on the already existing data in the duke.txt file, how do we ensure that the expected.txt file also gets updated to include that data?

**Comment by kokrui** [comment]
I personally chose to add a section to the `runtest.sh` script that:

- backs up any data previously in `duke.txt` into a separate temporary file
- let my text ui test run without a file / with no existing data
- restores the original duke.txt at the end of the test

**Comment by lamchenghou** [comment]
I agree with @kokrui, we can also dedicate a separate `txt` file for when we run `runtest.sh` (and purging it before every run). What I did was to allow `Duke` to access different files via different pathnames. 

I've also noticed that the relative pathnames resolve differently in `Duke.java` and when we use `runtest.sh`.

**Comment by eyelessrhyme7** [comment]
I open the scanner early on in my code, and I place the contents of my file in the INPUTS.txt within triple quotation marks """T|1|read book
D|0|return book|2023-09-06 18:30:00
D|0|submit quiz|2023-09-08 23:59:00
D|0|return book|2023-10-02 23:59:00
E|0|project meeting|2023-09-13 18:30:00|2023-09-13 20:30:00
T|1|join sports club"""". This way I can test out the script with a mock file, which is being passed as a part of the inputs and read from the Scanner. This approach will allow to include testing of the file reading/Storage class of our program.
